pkg/controller/clusterissuers/checks.go

```diff
@@ -39,7 +39,12 @@ func (c *controller) issuersForSecret(secret *corev1.Secret) ([]*v1alpha2.Cluste
 		}
 		if (iss.Spec.ACME != nil && iss.Spec.ACME.PrivateKey.Name == secret.Name) ||
 			(iss.Spec.CA != nil && iss.Spec.CA.SecretName == secret.Name) ||
-			(iss.Spec.Vault != nil && iss.Spec.Vault.Auth.TokenSecretRef.Name == secret.Name) {
+			(iss.Spec.Vault != nil && iss.Spec.Vault.Auth.TokenSecretRef != nil &&
+				iss.Spec.Vault.Auth.TokenSecretRef.Name == secret.Name) ||
+			(iss.Spec.Vault != nil && iss.Spec.Vault.Auth.Kubernetes != nil &&
+				iss.Spec.Vault.Auth.Kubernetes.SecretRef.Name == secret.Name) ||
+			(iss.Spec.Vault != nil && iss.Spec.Vault.Auth.AppRole != nil &&
+				iss.Spec.Vault.Auth.AppRole.SecretRef.Name == secret.Name) {
```

Comment:
Double check the issuers package too :)

Reply:
https://github.com/jetstack/cert-manager/blob/6bf7c0f079324c7f8cb13c3e71610e1074c97495/pkg/controller/issuers/checks.go#L50

Looks good